dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -121,6 +121,11 @@ EXPORTED_VARIABLES_ONLY_IN_VARS+=('TARGET_ARCH')
 EXPORTED_VARIABLES_ONLY_IN_VARS+=('TOOLCHAIN')
 EXPORTED_VARIABLES_ONLY_IN_VARS+=('WERROR')
 EXPORTED_VARIABLES_ONLY_IN_VARS+=('WPEDANTIC')
+EXPORTED_VARIABLES_ONLY_IN_VARS+=('CC[ =]' 'CXX' 'CCAS')
+EXPORTED_VARIABLES_ONLY_IN_VARS+=('AR[ =]' 'RANLIB')
+EXPORTED_VARIABLES_ONLY_IN_VARS+=('AS' 'NM' 'SIZE')
+EXPORTED_VARIABLES_ONLY_IN_VARS+=('LINK' 'LINKXX')
```

Comment:
`RANLIB`,  `LINKXX` and `NM` are not exported in vars, can you check if those exports are needed?

Reply:
nice catch :)

It seems RANLIB and NM are required in several places:

- openthread package:

https://github.com/RIOT-OS/RIOT/blob/48bdd7018a398e418a5cd134715d0bb9ed820f90/pkg/openthread/Makefile#L52

- generate-xcompile-toolchain.sh (only RANLIB):

https://github.com/RIOT-OS/RIOT/blob/48bdd7018a398e418a5cd134715d0bb9ed820f90/dist/tools/cmake/generate-xcompile-toolchain.sh#L22

- tests/libc_newlib (only NM):

https://github.com/RIOT-OS/RIOT/blob/48bdd7018a398e418a5cd134715d0bb9ed820f90/tests/libc_newlib/Makefile#L36-L37


LINKXX is only used in Makefile.include so it might not need to be exported at all.

Reply:
Yes I think it is probably not needed.

Reply:
I pushed a couple of commits to improve things. Not sure it's enough though ;)

Reply:
why are you not sure?

Reply:
> why are you not sure?

Maybe you'll find something, who knows ?